src/sentry/utils/snowflake.py

```diff
@@ -55,3 +67,28 @@ def snowflake_id_generation():
     ID_VALIDATOR.validate(snowflake_id)
 
     return snowflake_id
+
+
+def get_redis_cluster(redis_key: str):
+    return redis.clusters.get("default").get_local_client_for_key(redis_key)
+
+
+def get_sequence_value_from_redis(redis_key: str, starting_timestamp: int) -> Tuple[int, int]:
+    cluster = get_redis_cluster(redis_key)
+
+    for i in range(int(_TTL.total_seconds())):
+        timestamp = starting_timestamp - i
+
+        # We are decreasing the value by 1 each time since the incr operation in redis
+        # initializes the counter at 1. For our region sequences, we want the value to
+        # be from 0-15 and not 1-16
+        sequence_value = cluster.incr(timestamp)
+        sequence_value -= 1
+
+        if sequence_value == 0:
+            cluster.expire(timestamp, int(_TTL.total_seconds()))
+
+        if sequence_value < (1 << REGION_SEQUENCE.length):
```

Comment:
Can use the `MAX_AVAILABLE_REGION_SEQUENCES` constant here.

(Just for fun: if we were hacking on heavily optimized code, you could also write the whole thing as `if (sequence_value >> REGION_SEQUENCE.length) == 0:`. But that's not Pythonic at all.)